nomos-core/chain-defs/src/mantle/encoding.rs

```diff
@@ -237,21 +240,30 @@ fn decode_sdp_active(input: &[u8]) -> IResult<&[u8], SDPActiveOp> {
     // Metadata = UINT32 *BYTE
     let (input, declaration_id_bytes) = decode_hash32(input)?;
     let declaration_id = DeclarationId(declaration_id_bytes);
+
     let (input, nonce) = decode_uint64(input)?;
+
     let (input, metadata_len) = decode_uint32(input)?;
-    let (input, metadata_vec) =
-        map(take(metadata_len as usize), |b: &[u8]| b.to_vec()).parse(input)?;
+    let (input, metadata_bytes) = take(metadata_len as usize)(input)?;
+
+    let metadata = if metadata_bytes.is_empty() {
+        None
+    } else {
+        // Parse raw bytes into ActivityMetadata
+        match ActivityMetadata::from_metadata_bytes(metadata_bytes) {
+            Ok(meta) => Some(meta),
+            Err(_) => {
+                return Err(nom::Err::Error(Error::new(input, ErrorKind::Fail)));
+            }
```

Comment:
No need for the if/else, just this should be enough, also better to:
```suggestion
        let metadata = ActivityMetadata::from_metadata_bytes(metadata_bytes).map_err(|_| nom::Err::Error(Error::new(input, ErrorKind::Fail)))?;
```